Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -61,23 +61,19 @@ class HTTPClientTests: XCTestCase {
         let response = try httpClient.get(url: "http://localhost:\(httpBin.port)/get").wait()
         XCTAssertEqual(.ok, response.status)
     }
-
-    func testGetWithSharedEventLoopGroup() throws {
+    
+    func testGetWithDifferentEventLoopBackpressure() throws {
```

Comment:
Small question about this: without the changes (hopping the futures to the right event loop) this causes a TSan failure. I was surprised it was TSan that catched the threading issue and not a more direct check made by NIO. Am I doing something wrong or is this expected?

Reply:
I reckon the failure was accessing a property in HTTP client? NIO wouldn't be able to catch those. NIO can catch some regarding accessing the `context` from the wrong EventLoop (but not all). Do you have the TSan error report handy?

Reply:
Strange, commenting-out `hop(to:)` now causes the failure by `precondition(self.inEventLoop, file: file, line: line)`, but I guess there is some randomness and it might happen for TSan to be the first to detect the issue before NIO can even throw?

Reply:
ok, cool, thanks!

Reply:
@adtrevor yes, it's possible that TSan finds it before NIO

Reply:
the important thing is that for any `hop(to:)` there's at least one unit test that fails (even without TSan) if the `hop(to:)` is commented out.